### PR TITLE
Added WallClock class

### PR DESCRIPTION
**Pull request by sherman-the-tank**
To provide much faster xxxNowInSec() and xxxNowInMilliSec() methods

This class will replace TimeUtils class in a separate PR

### CODE REVIEW

**Comment by sherman-the-tank** [comment]
jenkins go

File: src/common/time/WallClock.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * The *fast* versions are way faster than the *slow* versions, but not as
+ * precise as the *slow* versions. Choose wisely
+ *
+ */
+class WallClock final {
+public:
+    WallClock() = delete;
+
+    static int64_t slowNowInSec();
```
Comment (dangleptr) [inline src/common/time/WallClock.h]: Do we need slowXXX any more?

Why not change TimeUtils Implementation directly?  Why use another interface?

Comment (sherman-the-tank) [reply src/common/time/WallClock.h]: The name ```TimeUtils``` is too general. It's better to make names more specific and meaningful. What we need here is to get the wall clock time

```slowXXX()``` is more accurate than ```fastXXX()```, but slower, so we keep them for scenarios that require accurate time

Comment (dangleptr) [reply src/common/time/WallClock.h]: > slowXXX() is more accurate than fastXXX(), but slower, so we keep them for scenarios that require accurate time

Sounds reasonable.  

> The name TimeUtils is too general. It's better to make names more specific and meaningful. What we need here is to get the wall clock time

We'd better use another name to avoid copyright infringement

**Comment by dutor** [comment]
Here is the benchmarking:

```
============================================================================
/home/dutor/Wdir/nebula/src/common/time/test/WallClockBenchmark.cpprelative  time/iter  iters/s
============================================================================
gettimeofday_get_msec                                       24.40ns   40.99M
systemclock_get_time_t                           129.21%    18.88ns   52.96M
wallclock_get_msec_slow                          129.91%    18.78ns   53.25M
wallclock_get_msec_fast                          172.47%    14.15ns   70.69M
----------------------------------------------------------------------------
gettimeofday_get_sec                                        17.21ns   58.10M
wallclock_get_sec_slow                            95.37%    18.05ns   55.41M
wallclock_get_sec_fast                           129.28%    13.31ns   75.12M
============================================================================
```

```
============================================================================
/home/dutor/Wdir/nebula/src/common/time/test/DurationBenchmark.cpprelative  time/iter  iters/s
============================================================================
stead_clock_timer                                           44.45ns   22.50M
duration_timer                                   170.50%    26.07ns   38.36M
============================================================================
```

The CPU module is:
```
Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz
```

File: src/common/time/WallClock.h (hunk 1 above)
Comment (sherman-the-tank) [reply src/common/time/WallClock.h]: WallClock is not copyrighted. It's been used in a lot of places

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,11 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Debug")
 endif(NOT CMAKE_BUILD_TYPE)
 
+if(NOT ${CMAKE_BUILD_TYPE} EQUAL "Debug")
```
Comment (dutor) [inline CMakeLists.txt]: It seems that `STREQUAL` must be used instead here.

Comment (sherman-the-tank) [reply CMakeLists.txt]: Good catch 👍

**Comment by whitewum** [comment]
> Excellent! 👍
> 
> I want to know the usage scenarios of inaccurate methods such as fastNowInSec?

In some case, time accuracy is not so important, just a timestamp is enough.

File: src/common/time/detail/TscHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        });
+    t.detach();
+}
+
+}  // Anonymous namespace
+
+
+volatile uint64_t readTsc() {
+#ifdef DURATION_USE_RDTSCP
+    uint32_t eax, ecx, edx;
```
Comment (laura-ding) [inline src/common/time/detail/TscHelper.cpp]: ecx is not use

**Comment by dutor** [comment]
Jenkins go!
